src/OpenDB/src/lefout/lefout.cpp

```diff
@@ -114,27 +147,43 @@ void lefout::writeHeader(dbLib* lib)
   if (right_bus_delimeter == 0)
     right_bus_delimeter = ']';
 
-  char hier_delimeter = lib->getHierarchyDelimeter();
-
   if (hier_delimeter == 0)
     hier_delimeter = '|';
 
-  fprintf(_out, "VERSION %s ;\n", tech->getLefVersionStr());
-  fprintf(_out,
-          "NAMESCASESENSITIVE %s ;\n",
-          tech->getNamesCaseSensitive().getString());
+  writeVersion(tech->getLefVersionStr());
+  writeNameCaseSensitive(tech->getNamesCaseSensitive());
+  writeBusBitChars(left_bus_delimeter, right_bus_delimeter);
+  writeDividerChar(hier_delimeter);
+  writePropertyDefinitions(lib);
+
+  if (lib->getLefUnits()) {
+    writeUnits(lib->getLefUnits());
+  }
+}
+
+void lefout::writeDividerChar(char hier_delimeter)
+{
+  fprintf(_out, "DIVIDERCHAR \"%c\" ;\n", hier_delimeter);
+}
+
+void lefout::writeUnits(int database_units)
+{
+  fprintf(_out, "UNITS\n");
+  fprintf(_out, "    DATABASE MICRONS %d ;\n", database_units);
+  fprintf(_out, "END UNITS\n");
+}
+
+void lefout::writeBusBitChars(char left_bus_delimeter, char right_bus_delimeter)
+{
   fprintf(_out,
           "BUSBITCHARS \"%c%c\" ;\n",
           left_bus_delimeter,
           right_bus_delimeter);
-  fprintf(_out, "DIVIDERCHAR \"%c\" ;\n", hier_delimeter);
-  writePropertyDefinitions(lib);
+}
 
-  if (lib->getLefUnits()) {
-    fprintf(_out, "UNITS\n");
-    fprintf(_out, "    DATABASE MICRONS %d ;\n", lib->getLefUnits());
-    fprintf(_out, "END UNITS\n");
-  }
+void lefout::writeNameCaseSensitive(const dbOnOffType on_off_type)
+{
+  fprintf(_out, "NAMESCASESENSITIVE %s ;\n", on_off_type.getString());
```

Comment:
Obsolete as of LEF 5.6

Reply:
Should this writer just write out LEF 5.8 regardless of the underlying tech version?